snap/local/hooks/cmd/configure/options.go

```diff
@@ -60,8 +61,59 @@ func options() {
 
 	hooks.Info("edgexfoundry:configure-options: handling config options for a single service: " + *service)
 
+	// process the EdgeX >=2.2 snap options
+	err = app_options.ProcessAppCustomOptions(*service)
+	if err != nil {
+		hooks.Error(fmt.Sprintf("edgexfoundry:configure-options: could not process custom options: %v", err))
+		os.Exit(1)
+	}
+
+	// process the legacy snap options
 	if err := applyConfigOptions(*service); err != nil {
 		hooks.Error(fmt.Sprintf("edgexfoundry:configure-options: error handling config options for %s: %v", *service, err))
 		os.Exit(1)
 	}
 }
+
+func processAppOptions() {
+	err := app_options.ProcessAppConfig(
+		"core-data",
+		"core-metadata",
+		"core-command",
+		"support-notifications",
+		"support-scheduler",
+		"app-service-configurable",
+		"device-virtual",
+		"security-secret-store",
+		"security-secretstore-setup",
+		"security-proxy-setup",
+		"security-bootstrapper",
+		"sys-mgmgt-agent",
```

Comment:
I'd suggest dropping this as the SMA is considered deprecated and will be removed completely in EdgeX 3.0.

Reply:
Combining the new options with the old ones is disallowed. As a result, it will not be possible for someone to configure SMA using the old options and everything else using the new ones. The only way to respect the deprecation policy is to keep this service functional (and configurable) until EdgeX 3.0 release, at which we can drop that too.

Reply:
I wasn't suggesting combining the options, I was just suggesting that you don't support the SMA at all with the new scheme. And afaik there's _nothing_ in the deprecation policy which states that new features must support deprecated services or applications, that would seem to defeat the purpose of deprecation IMHOP.

Reply:
As explained, this is necessary to allow using the new scheme with applications while still allowing the configuration of SMA. This isn't a new feature for SMA, but a new feature for the snap as a whole and that should not make the SMA incompatible with everything else when using the new scheme.